pkg/objstore/gcs/gcs_test.go

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package gcs
+
+import (
+	"context"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+
+	"github.com/go-kit/kit/log"
+
+	"github.com/thanos-io/thanos/pkg/testutil"
+)
+
+func TestBucket_Get_ShouldReturnErrorIfServerTruncateResponse(t *testing.T) {
+	srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
Does it make sense to put to acceptance test? 🤔

Reply:
The acceptance test uses the real S3/GCS backend. To simulate this case we need a mocked one. Am I missing anything, or maybe I don't understand what you mean?

Reply:
You are right, it's hard to mock, but that what I meant